Lib/test/support/__init__.py

```diff
@@ -2540,6 +2540,7 @@ def exceeds_recursion_limit():
 def without_optimizer(func):
     try:
         import _testinternalcapi
+        from _testinternalcapi import get_optimizer, set_optimizer
```

Comment:
This is sort of funky. Maybe replace remove the old import:

```suggestion
        from _testinternalcapi import get_optimizer, set_optimizer
```

...and use `get_optimizer` and `set_optimizer` directly below?